templates/rust/pkgs/default.nix

```diff
@@ -1,6 +1,22 @@
-{rustPlatform}:
+{
+  rustPlatform,
+  # you can add imports here
+}:
 rustPlatform.buildRustPackage rec {
   name = "my-package";
   src = ../.;
-  cargoLock.lockFile = ../Cargo.lock;
+  cargoLock = {
+    lockFile = ../Cargo.lock;
+    # The hash of each dependency that uses a git source must be specified
+    # The hash can be found by setting it to lib.fakeSha256 as shown below and running flox build.
+    # The build will fail but output the expected sha, which can then be added here.
+    # outputHashes = {
+    #   "dependency-0.0.0" = lib.fakeSha256;
+    # };
+  };
+  # non-Rust dependencies of your project can be added in buildInputs
+  # buildInputs =
+  #   [openssl]
+  # platform specific dependencies can be added as well
+  #   ++ lib.optional hostPlatform.isDarwin [darwin.apple_sdk.frameworks.Security];
```

Comment:
`libiconv` is pretty commonly referenced on macos too
We should also link to the different apple frameworks or explain how to read the linker error messages in the nix context

Reply:
I'm not incredibly familiar with all the different apple frameworks - would you be able to explain that? Or point me to some docs?

Reply:
Me neither my familiarity stems from trying to build rust projects and ever so often the linker fails
`Security` is a fairly frequent one (e.g. its a transitive dependency of the very popular tokio web-stack)

Reply:
Haha okay I'll poke around

Reply:
Added comments about linker failures. I think we a need a way for users to find frameworks, and I put a comment with desired behavior in the last commit, but that's not reality yet